File: Tree/101. Symmetric Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return isflipTree(root.left,root.right);

    }
    public boolean isflipTree(TreeNode p, TreeNode q){
        if(p==null&&q==null) return true;
        if(p==null||q==null) return false;
        if(p.val==q.val&&isflipTree(p.left,q.right)&&isflipTree(p.right,q.left)){
            return true;
        }
        return false;
    }
}

File: Tree/101. Symmetric Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root){
        return SymCompare(root->left,root->right);
    }
    bool SymCompare(TreeNode* p,TreeNode* q){
        if(p==nullptr||q==nullptr){
            if(p==q){
                return true;
            }else{
                return false;
            }
        }
        if(SymCompare(p->right,q->left)&&SymCompare(p->left,q->right)&&p->val==q->val){
            return true;
        }else{
            return false;
        }
    }
};